python/tskit/text_formats.py

```diff
@@ -0,0 +1,85 @@
+# MIT License
+#
+# Copyright (c) 2021 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Module responsible for working with text format data.
+"""
+import numpy as np
+
+import tskit
+
+
+def parse_fam(fam_path: str, tc: tskit.TableCollection):
```

Comment:
Most of our conversion routines return a new `TableCollection`, not modify an existing one. Is the thinking here that the dominant use case is adding individuals to an existing tree sequence?

Reply:
The TC manipulation routines, OTOH, modify a TC in-place. For those, we decided that we should impose the rule that table collection routines modify in-place, whereas tree sequence routines alway return a new TS instance (which they have to, because of immutability). So I wonder if Nik is right here, and any TC routines that take a TC as input should do an in-place modification?

Reply:
Yes, but an input TC isn't needed here. So maybe the right thing is to return the TS.

Reply:
Oh, I guess I didn't understand what the `tc: tskit.TableCollection` was doing in the function sig then, sorry - didn't look at the code itself.

Reply:
The original thinking was that this is only affecting the individual table, so it's not going to return a very interesting (or useful) TableCollection on its own. The only way you could use it is to either update the returned tables to set sequence length, add edges/nodes etc, or to copy the individual table that we create here.

There is an argument for just returning the IndividualTable I guess, which would clarify this?

Reply:
Ah I see. Returning an `IndividualTable` is conceptually clearer - but it is currently a bit clunky to use. You can't assign tables  to a `TableCollection` although I see no reason why you shouldn't be able to if we implemented it: https://github.com/tskit-dev/tskit/issues/1489